### PR TITLE
GetProcNFSDExport(mountpoint) retrieves the nfs export info for an expor...

### PR DESCRIPTION
**Pull request by wplu**
https://jira.zenoss.com/browse/CC-742

DEMO:
```
# plu@plu-9: go test -v
=== RUN TestGetProcNFSDExport
--- PASS: TestGetProcNFSDExport (0.00 seconds)
=== RUN TestGetProcNFSDExports
--- PASS: TestGetProcNFSDExports (0.00 seconds)
...
PASS
ok  	github.com/control-center/serviced/commons/proc	0.012s
```

### CODE REVIEW

**Comment by wplu** [comment]
```
=== RUN TestMonitorExportedVolume
I0123 13:55:19.758308 14069 nfsd_test.go:132] ==== writing to exports /tmp/tstproc341902113/fs/nfsd/exports: #
/exports/serviced_var   *(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
I0123 13:55:19.758749 14069 nfsd.go:147] monitoring exported volume /exports/serviced_var at polling interval: 2s
I0123 13:55:19.758785 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:19.758825 14069 nfsd.go:157] exported info has changed for mountpoint /exports/serviced_var
I0123 13:55:19.758912 14069 nfsd.go:168] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
I0123 13:55:21.759107 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:23.759324 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:25.758874 14069 nfsd_test.go:159] ==== writing to exports /tmp/tstproc341902113/fs/nfsd/exports: #
/exports/serviced_var   *(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
I0123 13:55:25.759520 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:25.759552 14069 nfsd.go:157] exported info has changed for mountpoint /exports/serviced_var
I0123 13:55:25.759608 14069 nfsd.go:168] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
I0123 13:55:27.759772 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:29.759986 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:31.759275 14069 nfsd_test.go:169] ==== clearing exports /tmp/tstproc341902113/fs/nfsd/exports: #
I0123 13:55:31.760176 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:31.760207 14069 nfsd.go:157] exported info has changed for mountpoint /exports/serviced_var
W0123 13:55:31.760234 14069 nfsd.go:161] volume /exports/serviced_var is not exported - further action may be required
I0123 13:55:33.760408 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:35.760629 14069 nfsd.go:151] determining export status for DFS NFS volume /exports/serviced_var
I0123 13:55:37.759939 14069 nfsd.go:176] no longer monitoring export status for DFS NFS volume /exports/serviced_var
--- PASS: TestMonitorExportedVolume (18.00 seconds)
```

**Comment by wplu** [comment]
TODO: pass in state changed function handler to check results

DEMO:
```
=== RUN TestMonitorExportedVolume
I0123 14:08:52.892267 24434 nfsd_test.go:132] ==== writing to exports /tmp/tstproc297655013/fs/nfsd/exports: #
/exports/serviced_var   *(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
I0123 14:08:52.892664 24434 nfsd.go:147] monitoring exported volume /exports/serviced_var at polling interval: 2s
I0123 14:08:52.892715 24434 nfsd.go:157] exported info has changed for mountpoint /exports/serviced_var
I0123 14:08:52.892777 24434 nfsd.go:168] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
I0123 14:08:58.892830 24434 nfsd_test.go:159] ==== writing to exports /tmp/tstproc297655013/fs/nfsd/exports: #
/exports/serviced_var   *(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
I0123 14:08:58.893284 24434 nfsd.go:157] exported info has changed for mountpoint /exports/serviced_var
I0123 14:08:58.893361 24434 nfsd.go:168] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
I0123 14:09:04.893190 24434 nfsd_test.go:169] ==== clearing exports /tmp/tstproc297655013/fs/nfsd/exports: #
I0123 14:09:04.893870 24434 nfsd.go:157] exported info has changed for mountpoint /exports/serviced_var
W0123 14:09:04.893919 24434 nfsd.go:161] volume /exports/serviced_var is not exported - further action may be required
I0123 14:09:10.893884 24434 nfsd.go:176] no longer monitoring export status for DFS NFS volume /exports/serviced_var
--- PASS: TestMonitorExportedVolume (18.00 seconds)
```

**Comment by wplu** [comment]
DEMO:
```
=== RUN TestMonitorExportedVolume
I0123 14:28:22.539556 08478 nfsd_test.go:132] ==== writing to exports /tmp/tstproc126182702/fs/nfsd/exports: #
/exports/serviced_var   *(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
I0123 14:28:22.539964 08478 nfsd.go:150] monitoring exported volume /exports/serviced_var at polling interval: 2s
I0123 14:28:22.540001 08478 nfsd.go:160] exported info has changed for mountpoint /exports/serviced_var
I0123 14:28:22.540051 08478 nfsd_test.go:152] ==== received exported volume info changed message - isExported:%!s(bool=true)
I0123 14:28:22.540073 08478 nfsd.go:174] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
I0123 14:28:28.540118 08478 nfsd_test.go:167] ==== writing to exports /tmp/tstproc126182702/fs/nfsd/exports: #
/exports/serviced_var   *(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
I0123 14:28:28.540640 08478 nfsd.go:160] exported info has changed for mountpoint /exports/serviced_var
I0123 14:28:28.540709 08478 nfsd_test.go:152] ==== received exported volume info changed message - isExported:%!s(bool=true)
I0123 14:28:28.540720 08478 nfsd.go:174] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
I0123 14:28:34.540477 08478 nfsd_test.go:178] ==== clearing exports /tmp/tstproc126182702/fs/nfsd/exports: #
I0123 14:28:34.541214 08478 nfsd.go:160] exported info has changed for mountpoint /exports/serviced_var
W0123 14:28:34.541263 08478 nfsd.go:164] volume /exports/serviced_var is not exported - further action may be required
I0123 14:28:34.541280 08478 nfsd_test.go:152] ==== received exported volume info changed message - isExported:%!s(bool=false)
I0123 14:28:40.541162 08478 nfsd.go:182] no longer monitoring export status for DFS NFS volume /exports/serviced_var
```

**Comment by wplu** [comment]
DEMO:
```
[root@jrivera-tb1 bin]# journalctl -u serviced -o cat -f
...
I0123 21:14:06.875211 13725 zzk.go:174] Starting a listener at /services
I0123 21:14:06.875417 13725 leader.go:190] calling wait on /storage/leader/_c_fffe0eb9b69e5bd2f7db614859237755-leader-0000000004: %!s(<nil>)
I0123 21:14:06.875446 13725 nfsd.go:150] monitoring exported volume /exports/serviced_var at polling interval: 30s
I0123 21:14:06.875482 13725 nfsd.go:160] exported info has changed for mountpoint /exports/serviced_var
I0123 21:14:06.875554 13725 nfsd.go:174] DFS NFS volume /exports/serviced_var (uuid:45a148e9:89326106:00000000:00000000) is exported
```

**Comment by wplu** [comment]
DEMO:
```
[root@jrivera-tb1 bin]# cat /proc/fs/nfsd/exports
# Version 1.1
# Path Client(Flags) # IPs
/exports/serviced_var	*(rw,insecure,no_root_squash,async,wdelay,nohide,no_subtree_check,uuid=45a148e9:89326106:00000000:00000000,sec=1)
[root@jrivera-tb1 bin]# systemctl status serviced
serviced.service - Zenoss ServiceD
   Loaded: loaded (/usr/lib/systemd/system/serviced.service; enabled)
   Active: active (running) since Thu 2015-01-22 14:12:14 CST; 24h ago
  Process: 16009 ExecStopPost=/opt/serviced/bin/serviced-systemd.sh post-stop (code=exited, status=0/SUCCESS)
  Process: 16090 ExecStartPre=/opt/serviced/bin/serviced-systemd.sh pre-start (code=exited, status=0/SUCCESS)
 Main PID: 16105 (serviced)
   CGroup: /system.slice/serviced.service
           └─16105 /opt/serviced/bin/serviced

[root@jrivera-tb1 bin]# systemctl stop nfs-server
[root@jrivera-tb1 bin]# cat /proc/fs/nfsd/exports 
# Version 1.1
# Path Client(Flags) # IPs
[root@jrivera-tb1 bin]# 


I0123 21:21:36.877762 13725 nfsd.go:160] exported info has changed for mountpoint /exports/serviced_var
W0123 21:21:36.877828 13725 nfsd.go:164] volume /exports/serviced_var is not exported - further action may be required
W0123 21:21:36.877838 13725 server.go:80] DFS NFS volume /exports/serviced_var is not exported

```

File: commons/proc/nfsd.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+// GetProcNFSDExports gets a map to the /proc/fs/nfsd/exports
+func GetProcNFSDExports() (map[string]ProcNFSDExports, error) {
+	// read in the file
+	data, err := ioutil.ReadFile(GetProcNFSDExportsFilePath())
+	if err != nil {
+		return nil, err
+	}
+
+	exports := make(map[string]ProcNFSDExports)
+	scanner := bufio.NewScanner(strings.NewReader(string(data)))
```
Comment (smousa) [inline commons/proc/nfsd.go]: I think you can do something like  scanner := bufio.NewScanner(bytes.NewReader(data))
http://golang.org/pkg/bytes/#NewReader

File: commons/proc/nfsd.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+	export, ok := exports[mountpoint]
+	if !ok {
+		return nil, ErrMountPointNotExported
+	}
+
+	return &export, nil
+}
+
+// GetProcNFSDExports gets a map to the /proc/fs/nfsd/exports
```
Comment (smousa) [inline commons/proc/nfsd.go]: Add comment with expected row input

File: commons/proc/nfsd.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+		changed, err := hasFileChanged(GetProcNFSDExportsFilePath(), &modtime)
+		if err != nil {
+			glog.Warningf("unable to determine whether mountpoint %s is exported: %s", mountpoint, err)
+		} else if changed {
+			glog.Infof("exported info has changed for mountpoint %s", mountpoint)
+			mountinfo, err := GetProcNFSDExport(mountpoint)
+			if err != nil {
+				if err == ErrMountPointNotExported {
+					glog.Warningf("volume %s is not exported - further action may be required", mountpoint)
+					changedFunc(mountpoint, false)
```
Comment (smousa) [inline commons/proc/nfsd.go]: the TODO below is going into the changedFunc, right?

File: commons/proc/nfsd.go (hunk 1 above)
Comment (wplu) [inline commons/proc/nfsd.go]: I would rather not change that at this critical juncture - this is mimicking all the other functions

File: commons/proc/nfsd.go (hunk 3 above)
Comment (wplu) [inline commons/proc/nfsd.go]: removed TODO - done

**Comment by wplu** [comment]
TODO: poll at 10 minute intervals, restart NFS if volume is not exported

In addition, use nfsd/ instead of nfs/ - the modtime are unreliable - make sure to read file each time polling is done

**Pull request merged**